### PR TITLE
Concurrency and fixes

### PR DESCRIPTION
**Pull request by FynnBe**

### CODE REVIEW

File: .github/scripts/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+bioimageio.spec==0.4.9.post5
+loguru==0.7.2
+minio==7.2.3
+packaging==23.2
+ruyaml==0.91.0
+ruyaml==0.91.0
```
Comment (jmetz) [inline .github/scripts/requirements.txt:6]: Duplicate

File: .github/scripts/test_dynamically.py [hunk 2]
```diff
@@ -6,10 +6,14 @@
...
-from tqdm import tqdm
 from update_log import add_log_entry
 
-tqdm.__init__ = partialmethod(tqdm.__init__, disable=True)  # silence tqdm
+try:
+    from tqdm import tqdm
+except ImportError:
+    pass
+else:
+    tqdm.__init__ = partialmethod(tqdm.__init__, disable=True)  # silence tqdm
```
Comment (jmetz) [inline .github/scripts/test_dynamically.py:16]: I might be misreading but this should fail if no tqdm, right?

**Review by jmetz — CHANGES_REQUESTED** [review]
Hey @FynnBe , mostly looks good except for a couple commented things that probably need to be changed.

Did you test with the test scripts?

File: .github/scripts/test_dynamically.py (hunk 2 above)
Comment (FynnBe) [reply .github/scripts/test_dynamically.py:16]: no on ImportError there is a `pass` the `else` runs if there was no exception. So this should be fine

Comment (jmetz) [reply .github/scripts/test_dynamically.py:16]: Oh right... I guess as it's only a one line "else" section would have probably just included it in the "try" but your way is more correct I expect

**Pull request merged**